src/server/routes/signIn.ts

```diff
@@ -291,17 +292,38 @@ const oktaSignInController = async (
     // we're authenticated track this metric
     trackMetric('OktaSignIn::Success');
 
-    // If the user was able to log in with Okta BUT NOT VALIDATED, we will still try and sign them in with Identity
-    // This is because we will initally block unvalidated users from getting Okta sessions, but still allow them
-    // Identity sessions.
-    // We use the groups api to get the group membership, rather than checking the email validated field because
-    // social users do not have the email validated flag set until after they login for the first time (but are in the email validated group)
-
     if (response._embedded?.user.id) {
+      // retrieve the user groups
       const groups = await getUserGroups(response._embedded.user.id);
+
+      // check if the user has their email validated based on group membership
       const emailValidated = groups.some(
         (group) => group.profile.name === 'GuardianUser-EmailValidated',
       );
+
+      // check the user password strength
+      const hasWeakPassword = await isBreachedPassword(password);
```

Comment:
I don't how precise we need the metrics to be but in case of error I think this defaults to false

Reply:
Since this is just for getting a comparison, and it's very rare that the breached password api to be down, this should be okay in this scenario. Long term it shouldn't affect our numbers.